### PR TITLE
Create Pages-3.0 Bundle To Handle New Version Level

### LINKED ISSUE #12387: Investigate testJspFeatureChange test for EE9 in the jsp.2.3_fat bucket
When this test is modified to run with jsp-3.0 and jsp-2.3 it fails because:

```
package com.ibm.ws.jsp23.webcontainerext;

import com.ibm.ws.jsp.webcontainerext.JspVersion;

public class JspVersionImpl implements JspVersion {

    public JspVersionImpl() {}

    @Override
    public String getJspVersionString() {
        return "2.3";
    }
}
```

We don't have a getJspVersionString() for 3.0.

1) Should we split the 2.3 factories out into their own project / bundle and only load with the 2.3 feature and then create a 3.0 factories bundle and update and then only load those with the jsp 3.0 feature?
2) Should we implement something like the ServletVersion or use the JavaEEVersion?
3) Can we do something with the transformer tool here?

### PR DESCRIPTION
**Pull request by volosied**
another approach to fix #12387

 The sole commit can be cleaned up to track the changes more easily, but here's an overview: 

The com.ibm.ws.jsp.2.3 contains two methods specific to the version of JSP loaded. 

- JspVersionImpl#getJspVersionString
- GeneratorUtilsExtImpl#getClassFileVersion

These should be changed for pages-3.0 for proper checks to occur. 

I had pulled out the `org.apache.jasper.el.*` classes out of `com.ibm.ws.jsp.2.3 `and used those to create  `com.ibm.ws.jsp.el `

Then I created  `io.openliberty.pages.internal.3.0` from a copy of the `com.ibm.ws.jsp.2.3 ` bundle and updated the jsp version to 3.0 in the two methods above. 

Both `com.ibm.ws.jsp.el ` and  `io.openliberty.pages.internal.3.0` are transformed to jakarta namespaces.

I don't really like this approach. We're creating a lot of new bundles unnecessarily in my opinion. I think the transformation done in PR #15319 is more better suited to fix this version issue. 
 
 This is still a draft as there could be more improvements to this PR.

### CODE REVIEW

File: dev/io.openliberty.pages.internal.3.0/src/io/openliberty/pages30/translator/visitor/generator/GeneratorUtilsExtImpl.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            writer.println("_jspx_fnmap = org.apache.jasper.runtime.ProtectedFunctionMapper.getInstance();");
+            writer.println("_jspx_fnmap.mapFunction(null, null, null, null,null);");
+            writer.println("}");
+            writer.println();
+        }
+    }
+
+    @Override
+    public String getClassFileVersion() {
+         return "3.0";
```
Comment (volosied) [inline dev/io.openliberty.pages.internal.3.0/src/io/openliberty/pages30/translator/visitor/generator/GeneratorUtilsExtImpl.java:94]: Update Version Here

File: dev/io.openliberty.pages.internal.3.0/src/io/openliberty/pages30/webcontainerext/JspVersionImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.ibm.ws.jsp.webcontainerext.JspVersion;
+
+public class JspVersionImpl implements JspVersion {
+
+    public JspVersionImpl() {}
+
+    @Override
+    public String getJspVersionString() {
+         return "3.0";
```
Comment (volosied) [inline dev/io.openliberty.pages.internal.3.0/src/io/openliberty/pages30/webcontainerext/JspVersionImpl.java:21]: Updated Here.

File: dev/io.openliberty.pages.internal.3.0/bnd.bnd [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+Bundle-SymbolicName: io.openliberty.pages.internal.3.0
+Bundle-Description: Pages 3.0 Internal Bundle
```
Comment (volosied) [inline dev/io.openliberty.pages.internal.3.0/bnd.bnd:15]: New Bundle

File: dev/com.ibm.ws.jsp.el/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+Bundle-SymbolicName: com.ibm.ws.jsp.el
+Bundle-Description: EL Handing Specific for JSP/Pages
```
Comment (volosied) [inline dev/com.ibm.ws.jsp.el/bnd.bnd:15]: New Bundle. 

Used by both jsp-2.3 & pages-3.0

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
Builds fails due to:
`/home/jazz_build/_IJwGoEkREeu3FvTQ3tOvAg-EBC.PROD.WASRTC-8LJ-000-00-00/jbe/build/WS-CD-Open/dev/ant_build/resources/build_imports/liberty-macros.xml:967: Parallel execution timed out`

Not sure why?

**Comment by volosied** [comment]
#build

**Comment by wraschke** [comment]
#run-libby

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
Some EL Tests are encountering this error in jsp-2.3 (not pages-3.0) . Not sure why? This error is not occurring locally.  I downloaded the wlp from the build, and moved my com.ibm.jsp.el and com.ibm.ws.jsp.2.3 jar over.  Still encountered the same error.  

```
Exception thrown by application class 'javax.el.BeanELResolver$BeanProperties.get:271'
javax.el.PropertyNotFoundException: Property 'out' not found on type javax.el.ELClass
```

Running another build to see if it reoccurs.

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build  #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat, com.ibm.ws.jsf.2.2_fat, com.ibm.ws.jsf.2.3_fat, com.ibm.ws.webcontainer.servlet.4.0_fat, com.ibm.ws.jsp_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat, com.ibm.ws.jsf.2.2_fat, com.ibm.ws.jsf.2.3_fat, com.ibm.ws.webcontainer.servlet.4.0_fat, com.ibm.ws.jsp_fat

**Pull request closed without merging**